native/packages/react-native-bpk-component-button/src/layout/BpkButtonContainer.ios.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  View,
+  TouchableHighlight,
+} from 'react-native';
+import React from 'react';
+import PropTypes from 'prop-types';
+import LinearGradient from 'react-native-linear-gradient';
+
+import {
+  getStyleForElement,
+  getThemingForElement,
+  getGradientColors,
+  themePropType,
+} from './../utils';
+import styles from './../BpkButton-styles';
+
+const BpkButtonContainer = (props) => {
+  const theme = props.theme;
+  const gradientColors = getGradientColors(theme, props);
+  const containerStyle = getStyleForElement('container', props);
+  const buttonStyle = getStyleForElement('button', props);
+  const buttonTheme = getThemingForElement('button', theme, props);
+  const {
+    accessibilityLabel,
+    children,
+    disabled,
+    iconOnly,
+    onPress,
+    style,
+    title,
+    ...rest
+   } = props;
+  const accessibilityTraits = ['button'];
+
+  if (disabled) {
+    accessibilityTraits.push('disabled');
+  }
+  return (
+    <LinearGradient
+      colors={gradientColors}
+      style={[containerStyle, style]}
+    >
+      <TouchableHighlight
+        accessibilityComponentType="button"
+        accessibilityLabel={accessibilityLabel || title}
+        accessibilityTraits={accessibilityTraits}
+        disabled={disabled}
+        onPress={onPress}
+        style={[buttonStyle, buttonTheme]}
+        underlayColor={styles.underlayColor}
+        {...rest}
+      >
+        {children}
+      </TouchableHighlight>
+    </LinearGradient>);
+};
+
+BpkButtonContainer.propTypes = {
```

Comment:
Should we have a test that ensures these two have the same `PropTypes` exported?

Reply:
I would agree usually, but in this case there is a `large` prop used by iOS but not android

Reply:
We could take a similar approach as we do with tokens where we support platform specific tokens.

Reply:
like https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/src/react-native-output-test.js

Reply:
Yep

Reply:
that's a really good idea, I want to improve the code coverage anyway, I'll add that in!